### DISCUSSION TITLE
Submodule update that triggers a package-lock.json update

### DISCUSSION THREAD

**Discussion by maartenvansambeek** [renovatebot/renovate#27881] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Azure Pipelines

### Please tell us more about your question or problem

Hi,

I have a repo that contains an npm workspace that looks like this:

- `/package.json`
- `/package-lock.json`
- `/src/project/package.json`
- `/submodule/project2/package.json`

The second project (`/submodule/project2`) is in a submodule.

The `/package.json` references both packages as a workspace:

```json
{
  "workspaces": [
    "src/project",
    "submodule/project2",
  ]
}
```

Say the submodule is updated, and a new or updated dependency is added to it. When I update the submodule and run `npm install` manually, the package-lock.json is updated in the root `package-lock.json`.

If i use the `git-submodules` manager, I get a PR for the updated submodule, but I was hoping that it would also update my package-lock.json in the PR, so it wouldn't break the build that runs `npm ci`.

Is a setup like this possible? I know a better setup would be just publishing `project2` and referencing it, but I am not able to change the structure of the repository at the moment.

Kind regards,

Maarten

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment, marked as answer]
You could try with `postUpgradeTasks`

**Comment by maartenvansambeek** [comment]
Thank you very much, that fixed my problem!

I'm really impressed with all the configuration options available.
